light-zk.js/src/utxo.ts

```diff
@@ -630,27 +635,54 @@ export class Utxo {
         "aes-256-cbc",
         true,
       );
+
+      prefix = this.account.generateUtxoPrefixHash(commitment, PREFIX_LENGTH);
+
       if (!compressed) return ciphertext;
       const padding = sha3_256
         .create()
         .update(Uint8Array.from([...nonce, ...bytes_message]))
         .digest();
 
       // adding the 8 bytes as padding at the end to make the ciphertext the same length as nacl box ciphertexts of 120 bytes
-      return Uint8Array.from([...ciphertext, ...padding.subarray(0, 8)]);
+      return Uint8Array.from([
+        ...prefix,
+        ...ciphertext,
+        ...padding.subarray(0, 8),
+      ]);
     }
   }
 
+  /**
+   * @description Checks the UTXO prefix hash of PREFIX_LENGTH-bytes
+   *
+   * @returns {boolean} true || false
+   */
+  static checkPrefixHash({
+    account,
+    commitment,
+    prefix,
+  }: {
+    account: Account;
+    commitment: Uint8Array;
+    prefix: Uint8Array;
+  }): boolean {
+    return (
+      account.generateUtxoPrefixHash(commitment, PREFIX_LENGTH).join(",") ===
+      prefix.join(",")
```

Comment:
interesting :D
did you consciously decide to use join over toString()?

Reply:
yeah I did. *.join is faster:

<img width="1088" alt="Screenshot 2023-07-27 at 09 51 40" src="https://github.com/Lightprotocol/light-protocol/assets/7057784/e8fdc48b-6bb1-4dcc-b61f-f7062da54161">

Reply:
nice :D